### PR TITLE
Simplify visualization JS interface

### LINKED ISSUE #1994: Simplify all the visualizations!
On the javascript side, a visualization definition should be as simple as defining a render function that receives the slice object and the data. Currently it's a js object that exposes a `render` and a `resize` method, and `render` is expected to fetch the data on its own.

In this proposed change, the data fetching should be handled by the "controller", not by the visualization itself.

### PR DESCRIPTION
**Pull request by mistercrunch**
fixes https://github.com/airbnb/superset/issues/1994

Previously viz on the javascript side would be defined as an object exposing a `render` and a `resize` method, and would be expected to fetch the data asynchronously from the backend on their own. 

After this PR, viz will be a simple function that receives a payload (the response from the backend) as well as a slice object, and are expected to mutate the dom all they want. The framework's container objects are now in charge of fetching the data and handling success/errors.

A side mission was around refactoring the ChartContainer object by moving the querying into a redux action creator, and simplifying the lifecycle of the component quite a bit. For clarity, we now put the queryResponse in the state.

Note that this PR gets rid of 3-ish hundred lines of code.

Still TODO:

* fix some bugs in V2 around height handling
* figure out how to handle filter_box's special handling overriding

### CODE REVIEW

File: superset/assets/javascripts/explorev2/components/ChartContainer.jsx [hunk 1]
```diff
@@ -37,125 +33,80 @@ const propTypes = {
...
-        let payload = { error: msg };
-        try {
-          payload = JSON.parse(msg);
-        } catch (e) {
-          // pass
-        }
-        props.actions.chartUpdateFailed(payload.error, payload.query);
-      },
-
+      error() {},
```
Comment (ascott) [inline superset/assets/javascripts/explorev2/components/ChartContainer.jsx]: airbnb javascript style guide calls for spaces after blocks and before the next statement. https://github.com/airbnb/javascript#whitespace--after-blocks

File: superset/assets/javascripts/explorev2/index.jsx [hunk 2]
```diff
@@ -32,7 +32,8 @@ const bootstrappedState = Object.assign(
     user_id: bootstrapData.user_id,
     chartUpdateStartTime: now(),
     chartUpdateEndTime: null,
-    chartStatus: 'loading',
+    chartStatus: 'trigger',
```
Comment (ascott) [inline superset/assets/javascripts/explorev2/index.jsx]: what does `trigger` mean?

File: superset/assets/webpack.config.js [hunk 3]
```diff
@@ -12,15 +12,15 @@ const VERSION_STRING = JSON.parse(fs.readFileSync('package.json')).version;
...
+    // 'css-theme': APP_DIR + '/javascripts/css-theme.js',
+    // common: APP_DIR + '/javascripts/common.js',
     dashboard: ['babel-polyfill', APP_DIR + '/javascripts/dashboard/Dashboard.jsx'],
     explore: ['babel-polyfill', APP_DIR + '/javascripts/explore/explore.jsx'],
     explorev2: ['babel-polyfill', APP_DIR + '/javascripts/explorev2/index.jsx'],
-    sqllab: ['babel-polyfill', APP_DIR + '/javascripts/SqlLab/index.jsx'],
-    standalone: ['babel-polyfill', APP_DIR + '/javascripts/standalone.js'],
-    welcome: ['babel-polyfill', APP_DIR + '/javascripts/welcome.js'],
-    profile: ['babel-polyfill', APP_DIR + '/javascripts/profile/index.jsx'],
+    // sqllab: ['babel-polyfill', APP_DIR + '/javascripts/SqlLab/index.jsx'],
```
Comment (ascott) [inline superset/assets/webpack.config.js]: curious why/when we deleted these lines and why they are commented out now. seems to me like we need these lines, so i'm confused why we would delete or comment out...

File: superset/assets/javascripts/explore/components/ExploreActionButtons.jsx [hunk 4]
```diff
@@ -17,9 +17,7 @@ export default function ExploreActionButtons({ canDownload, slice, query }) {
   return (
     <div className="btn-group results" role="group">
       <URLShortLinkButton slice={slice} />
-
```
Comment (ascott) [inline superset/assets/javascripts/explore/components/ExploreActionButtons.jsx]: these extra lines make the code more readable IMO. seems like an unnecessary change in this PR.

File: superset/assets/javascripts/explorev2/components/ChartContainer.jsx [hunk 5]
```diff
@@ -37,125 +33,80 @@ const propTypes = {
...
       d3format: (col, number) => {
         // mock d3format function in Slice object in superset.js
         const format = props.column_formats[col];
         return d3format(format, number);
       },
+      data: {
+        csv_endpoint: this.props.queryResponse.csv_endpoint,
```
Comment (ascott) [inline superset/assets/javascripts/explorev2/components/ChartContainer.jsx]: looks like `this.props` can be replaced with just `props` in this context.

File: superset/assets/javascripts/explore/components/ExploreActionButtons.jsx (hunk 4 above)
Comment (mistercrunch) [inline superset/assets/javascripts/explore/components/ExploreActionButtons.jsx]: I'm just trying to be consistent with the rest of the codebase where we do not typically do this. In the absence of an enforced linting rule I'd go for consistency. I'm fine removing this from this PR.

File: superset/assets/javascripts/explorev2/components/ChartContainer.jsx (hunk 1 above)
Comment (mistercrunch) [inline superset/assets/javascripts/explorev2/components/ChartContainer.jsx]: Alright I can change this. I wish the linter would enforce this so we wouldn't have to go back and forth on it.

File: superset/assets/javascripts/explorev2/index.jsx (hunk 2 above)
Comment (mistercrunch) [inline superset/assets/javascripts/explorev2/index.jsx]: oops, that was leftover from a previous approach before I decided to use an action creator to run queries, removing...

File: superset/assets/webpack.config.js (hunk 3 above)
Comment (mistercrunch) [inline superset/assets/webpack.config.js]: Oh this is my "optimization" hack to get faster build times while iterating, I'll have to remove before merging for sure!

**Comment by mistercrunch** [comment]
@ascott , thanks for the review! I'm ready for another round if needed.

File: superset/assets/javascripts/modules/superset.js [hunk 6]
```diff
@@ -222,14 +222,19 @@ const px = function () {
         timer = setInterval(stopwatch, 10);
         $('#timer').removeClass('label-danger label-success');
         $('#timer').addClass('label-warning');
-        this.viz.render();
+        $.getJSON(this.jsonEndpoint(), queryResponse => {
+          try {
+            vizMap[data.form_data.viz_type](this, queryResponse);
+            this.done(queryResponse);
```
Comment (ascott) [inline superset/assets/javascripts/modules/superset.js]: curious about the case when `this.done` would fail, which makes the `try/catch` neccessary... could you elaborate?

**Comment by ascott** [comment]
LGTM!
